### DISCUSSION TITLE
Automating updates of a cask I maintain the app for?

### DISCUSSION THREAD

**Discussion by pianogamer5** [Homebrew/discussions#3808] [Tap maintenance and Homebrew development]
Hey all,

I work for TechSmith, makers of Snagit and Camtasia. At some point late last year, we found out that there were a community-maintained casks of both of those apps. Once we discovered that, I was curious about what we could do to automate the process of updating these casks within our Jenkins build pipeline. Is this something that has been done already for other casks, or are there any guidelines/best practices on how to do this?

If this is uncharted territory, I've got a couple questions about how best to proceed. 
* Ideally, it'd be great to use `bump-cask-pr` (which is actually something I've started doing manually every time we update Snagit, the product I specifically work on), but I'm not sure that would work out well, as `bump-cask-pr` appears to assume the new version is live, which might not be the case when our Jenkins release scripts finish running due to CDNs needing to be flushed (currently a manual step). 
* If `bump-cask-pr` is not an option, then I'd probably make our own script that replicates what `bump-cask-pr` does, which AFAICT is the following:
1. Update a fork of the `homebrew-cask` repository
2. Update `version` in the cask file
3. Calculate a `sha256` hash of our DMG and add that to the cask file
4. Commit the changes to the cask file to the fork of `homebrew-cask`
5. Open a PR of the fork against the upstream `homebrew-cask` repository with the changes

    Is that an accurate understanding of how things work?

* There is also something I've noticed that might make all of this irrelevant: Sparkle support. For Camtasia and Snagit we use Sparkle to push out updates, and I noticed while looking through the documentation that [`livecheck` supports Sparkle as a strategy.](https://docs.brew.sh/Brew-Livecheck#sparkle-strategy-block) How exactly does that work? There's currently a `livecheck` block in the Snagit and Camtasia casks (oddly, they're different), but they don't use Sparkle and still require manual intervention to update. It'd be neat if we could put our Sparkle feed URLs in there and never have to touch this cask again for future updates (at least, so long as our Sparkle feed URL stays the same), but admittedly I don't know enough about how Sparkle or Ruby works to do that entirely on my own, and I also don't know if that's how `livecheck` is intended to function.

Thanks in advance for any tips on this endeavor! Also, on a personal note, thanks for Homebrew and Homebrew Cask, if any maintainers/contributors are reading this, I've been using it for a long time, well before I even started working for TechSmith, and it's been super great, as someone who also really likes using Linux and the package manager paradigm on Linux distros. Keep up the great work!

**Comment by p-linnane** [comment]
Some projects run `bump-cask-pr` upon release in order to automatically open PR's. We see that pretty regularly. There are some GitHub Actions floating around that do it.

We can utilize the Sparkle `livecheck` strategy, but the casks will still require PR's to update. It doesn't happen automatically.

**Reply by pianogamer5** [reply]
Thanks for the reply! Is there an example of implementing the Sparkle `livecheck` strategy I could use? Also how exactly does livecheck work?

**Reply by SMillerDev** [reply]
https://docs.brew.sh/Brew-Livecheck

**Comment by p-linnane** [comment]
Documentation on `livecheck` can be found [here](https://docs.brew.sh/Brew-Livecheck). The Sparkle strategy specifically is [here](https://docs.brew.sh/Brew-Livecheck#sparkle-strategy-block).

You can use [rectangle.rb](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/rectangle.rb) as an example of a basic Sparkle `livecheck`.
